### ISSUE TITLE
html5: sapp_quit

### ISSUE THREAD

**Issue by caiiiycuk** [floooh/sokol-samples#55]
Hi. Looks like `sapp_quit` is not working for html5 backend. I have a test suite for js-dos, that runs new instances of js-dos one by one on same canvas. Everything works, but previous instances of js-dos tries to update canvas even if they are stopped. To stop them I used `sapp_quit`.

It seems that sapp_quit is no sense for html backend. I think it should:
* unregister all events listeners
* unregister raf callback

OT: How you develop sokol_app project? For me it's very hard to find functions relative to emscripten implementation. I using just emacs, can you give any hints? Thanks!

**Comment by floooh** [comment]
Yes indeed, the sokol-app HTML5 code doesn't have a "cleanup path", that's because a webpage can "stop existing" at any time, so for the normal use case it doesn't really make sense to run through shutdown code (I haven't anticipated your use case).

The quit behaviour is described here: https://github.com/floooh/sokol/blob/82707c2dd90b4086341c7e4171909cf0bb2753bb/sokol_app.h#L388-L451

sapp_quit() itself should do the cleanup btw, it only sets a flag:
https://github.com/floooh/sokol/blob/82707c2dd90b4086341c7e4171909cf0bb2753bb/sokol_app.h#L7431 (the other important flag is "quit_requested").

The various platform's code paths then react differently to this flag, some send a window-close message to the operating system, some use this as a flag to exit the frame loop.

As for how to work on the headers: I don't have a specific editor preference. For emscripten stuff I usually work in VSCode and use on the of the ```*-vscode-*``` build configs. The platform-specific functions usually have a special prefix in function- and struct-names one can search for (e.g. ```_emsc_```, ```_win32_```, ```_gl_```, ```_d3d11_``` etc...). And I often have 2 views open in the editor on the same file. But that's basically it.

(I'll move the ticket over to the sokol-repository, but I'm currently deep in the WebGPU backend and won't start working on it yet).

For the implementation I would probably add the following things:

- an ```sapp_desc.html5_enable_shutdown``` flag (or similar name), which tells sokol-app that it needs to run in a mode where shutdown/cleanup is enabled
- a ```_sapp_emsc_shutdown()``` function which does all the emscripten-specific shutdown work
- in the emscripten frame callback, add code to check the quit-ordered flag, and if this and ```sapp_desc.html5_enable_shutdown``` is set, call the _sapp_emsc_shutdown() function

PS: I'll move the ticket over to the sokol repository
